autoload/flutter.vim

```diff
@@ -45,6 +45,20 @@ function! flutter#_exit_cb(job, status) abort
   call job_stop(a:job)
 endfunction
 
+function! flutter#_on_exit_nvim(job_id, data, event) abort dict
+  if exists('g:flutter_job')
+    unlet g:flutter_job
+  endif
+  call jobstop(a:job_id)
+endfunction
+
+function! flutter#_on_output_nvim(job_id, data, event) abort dict
+  if !empty(a:data)
+    let lines = filter(a:data, '!empty(v:val)')
+    call nvim_buf_set_lines(bufnr('__Flutter_Output__'), -1, -1, v:true, lines)
```

Comment:
I'm not sure what happens (or is supposed to happen here) when getting data that doesn't end on newline boundaries.

Reply:
In Vim8, channels for jobs default to reading on newlines. Is it the same in NV?

Reply:
Apparently not!

> Note 2:
>	Job event handlers may receive partial (incomplete) lines. For a given
>	invocation of on_stdout/on_stderr, `a:data` is not guaranteed to end
>	with a newline.
>	  - `abcdefg` may arrive as `['abc']`, `['defg']`.
>	  - `abc\nefg` may arrive as `['abc', '']`, `['efg']` or `['abc']`,
>	    `['','efg']`, or even `['ab']`, `['c','efg']`.
https://neovim.io/doc/user/job_control.html

Reply:
Actually, I don't necessarily think it would matter, as long as it just writes to the `__Flutter_Output__` buffer?